numpy/_core/src/umath/string_buffer.h

```diff
@@ -1533,6 +1533,46 @@ string_expandtabs(Buffer<enc> buf, npy_int64 tabsize, Buffer<enc> out)
     npy_intp new_len = 0, line_pos = 0;
 
     Buffer<enc> tmp = buf;
+
+    if (enc == ENCODING::UTF8) {
```

Comment:
I'm confused why you're only doing this for UTF-8. I would naively expect it to be possible to update this generically without adding this if statement.

Reply:
To quote my original PR description:

> Why only UTF-8? I had a bit of trouble understanding the whole machinery involved, but it looks like only UTF-8 re-encodes on buffered memset, so it’s the only encoding that benefits from clustering. When I tried expanding the approach to the other encodings, it actually slowed down. I might be missing something, though.

Reply:
Ah yeah, sorry, the character counting is off with my thought of course :facepalm: